models/concerns/validate_data.rb

```diff
@@ -107,6 +108,24 @@ def check_matrices
     end
   end
 
+  def check_rests
+    used_rest_ids = @hash[:vehicles].flat_map{ |v| v[:rest_ids] }
+    used_rest_ids.compact!
+
+    used_rest_ids.uniq.each{ |rest_id|
+      corresponding = @hash[:rests].find{ |r| r[:id] == rest_id }
+      unless corresponding
+        raise OptimizerWrapper::DiscordantProblemError.new(
+          "Unknown rest provided to at least one vehicle : #{rest_id}"
+        )
+      end
+
+      next unless corresponding[:timewindows].to_a.size > 1
```

Comment:
This is handled by active_hash (AH) during object creation automatically and AH sends the indices for finding the offending parameter easily in a uniform manner (with an IDerror if I remember correctly). 
```suggestion
      next unless corresponding && corresponding[:timewindows].to_a.size > 1
```